src/sas/mod.rs

```diff
@@ -0,0 +1,190 @@
+    W = b'W',
+    /// A missing value represented using the ASCII letter 'X'.
+    X = b'X',
+    /// A missing value represented using the ASCII letter 'Y'.
+    Y = b'Y',
+    /// A missing value represented using the ASCII letter 'Z'.
+    Z = b'Z',
+}
+
+impl MissingValue {
+    /// Return the ASCII code representation for the missing value.
+    #[inline]
+    pub const fn code(&self) -> u8 {
+        *self as u8
+    }
+}
+
+impl TryFrom<u8> for MissingValue {
+    type Error = &'static str;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            b'.' => Ok(MissingValue::Period),
+            b'_' => Ok(MissingValue::Underscore),
+            b'A' => Ok(MissingValue::A),
+            b'B' => Ok(MissingValue::B),
+            b'C' => Ok(MissingValue::C),
+            b'D' => Ok(MissingValue::D),
+            b'E' => Ok(MissingValue::E),
+            b'F' => Ok(MissingValue::F),
+            b'G' => Ok(MissingValue::G),
+            b'H' => Ok(MissingValue::H),
+            b'I' => Ok(MissingValue::I),
+            b'J' => Ok(MissingValue::J),
+            b'K' => Ok(MissingValue::K),
+            b'L' => Ok(MissingValue::L),
+            b'M' => Ok(MissingValue::M),
+            b'N' => Ok(MissingValue::N),
+            b'O' => Ok(MissingValue::O),
+            b'P' => Ok(MissingValue::P),
+            b'Q' => Ok(MissingValue::Q),
+            b'R' => Ok(MissingValue::R),
+            b'S' => Ok(MissingValue::S),
+            b'T' => Ok(MissingValue::T),
+            b'U' => Ok(MissingValue::U),
+            b'V' => Ok(MissingValue::V),
+            b'W' => Ok(MissingValue::W),
+            b'X' => Ok(MissingValue::X),
+            b'Y' => Ok(MissingValue::Y),
+            b'Z' => Ok(MissingValue::Z),
+            _ => Err("Not a missing value"),
+        }
+    }
+}
+
+/// A 64-bit SAS® floating point number.
+///
+/// SAS® floating point numbers are based on IBM floating point numbers, with
+/// different representations for missing values. When converted to an IEEE-754
+/// floating point number, the missing value representation is encoded as a
+/// valid NaN encoding.
+#[derive(Debug, Copy, Clone)]
+pub struct F64(super::F64);
+
+impl F64 {
+    /// Create a floating point value from its representation as a byte array in big endian.
+    ///
+    /// ```
+    /// use ibmfloat::sas::F64;
+    ///
+    /// let foreign_float = F64::from_be_bytes([0x2E, 0x00, 0, 0, 0, 0, 0, 0]);
+    ///
+    /// assert_eq!(foreign_float.to_bits(), 0x2E000000_00000000);
+    ///
+    /// let native_float = f64::from(foreign_float);
+    /// assert!(native_float.is_nan());
+    /// ```
+    #[inline]
+    pub const fn from_be_bytes(bytes: [u8; 8]) -> Self {
+        Self(super::F64::from_be_bytes(bytes))
+    }
+
+    /// Transmute a native-endian `u64` into an `F64`.
+    #[inline]
+    pub const fn from_bits(bits: u64) -> Self {
+        Self(super::F64::from_bits(bits))
+    }
```

Comment:
There's probably more methods on `super::F64` I need to expose on `sas::F64`. Just let me know.